src/components/MoneyRequestReportView/MoneyRequestReportView.tsx

```diff
@@ -182,40 +218,27 @@ function MoneyRequestReportView({report, policy, reportMetadata, shouldDisplayRe
                         }}
                     />
                 ) : (
-                    <MoneyReportHeader
-                        report={report}
-                        policy={policy}
-                        reportActions={reportActions}
-                        transactionThreadReportID={transactionThreadReportID}
-                        shouldDisplayBackButton
-                        onBackButtonPress={() => {
-                            if (!backToRoute) {
-                                goBackFromSearchMoneyRequest(activeWorkspaceID);
-                                return;
-                            }
-                            Navigation.goBack(backToRoute);
-                        }}
-                    />
+                    reportHeaderView
                 )}
                 <View style={[styles.overflowHidden, styles.flex1]}>
-                    {isTransactionThreadView ? (
-                        <ReportActionsView
+                    {shouldDisplayMoneyRequestActionsList ? (
+                        <MoneyRequestReportActionsList
                             report={report}
+                            policy={policy}
+                            transactions={transactions}
                             reportActions={reportActions}
-                            isLoadingInitialReportActions={reportMetadata?.isLoadingInitialReportActions}
-                            hasNewerActions={hasNewerActions}
                             hasOlderActions={hasOlderActions}
-                            parentReportAction={parentReportAction}
-                            transactionThreadReportID={transactionThreadReportID}
+                            hasNewerActions={hasNewerActions}
                         />
                     ) : (
-                        <MoneyRequestReportActionsList
+                        <ReportActionsView
                             report={report}
-                            policy={policy}
-                            transactions={transactions}
                             reportActions={reportActions}
-                            hasOlderActions={hasOlderActions}
+                            isLoadingInitialReportActions={reportMetadata?.isLoadingInitialReportActions}
                             hasNewerActions={hasNewerActions}
+                            hasOlderActions={hasOlderActions}
+                            parentReportAction={parentReportAction}
+                            transactionThreadReportID={transactionThreadReportID}
                         />
```

Comment:
Might want to consider doing the same as with the `reportHeaderView` and put it in the useMemo above

Reply:
yeah I guess I could, they accept so many props that I kinda wanted to left them here as these are the most important parts of `return`, and header is kind-of an addition.
Don't have strong opinions on this though